public/ipython/services/kernels/kernel.js

```diff
@@ -971,10 +972,20 @@ define([
         if (!_.isEmpty(msg.content.thrown)) {
             st = ". Stacktrace:\n" + msg.content.thrown.join("\n")
         }
-        // hide in console using filter: ^[^(?:Server log>)]
-        console[msg.content.level.toLowerCase()](
-            "Server log> ["+new Date(msg.content.time_stamp)+"] [" + msg.content.logger_name + "] " + msg.content.message + st
-        );
+        var lid = _.uniqueId("ui-logs-");
+        var detailedError = lid+"> ["+new Date(msg.content.time_stamp)+"] [" + msg.content.logger_name + "] " + msg.content.message + st;
+        // hide in console using filter: ^[^(?:ui-logs)]
+        console[msg.content.level.toLowerCase()](detailedError);
+        if (msg.content.level.toLowerCase() == "error") {
+            var logsPanel = $("#logsPanel");
+            var errorMsg = $("<li class='text text-danger' title='console log id: "+lid+". Click to print again'>[" +
+                                msg.content.logger_name + "] " + msg.content.message + "</li>");
+            errorMsg.click(function() {
+                console[msg.content.level.toLowerCase()](detailedError);
```

Comment:
unless someone has very wide screen, one would not see console & left-panel at the same time.
it should display the detailed error instead (maybe modal dialog).